src/Kmd.Logic.Gateway.Automation/PreValidation/ApiPreValidation.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Threading.Tasks;
+using Kmd.Logic.Gateway.Automation.Gateway;
+
+namespace Kmd.Logic.Gateway.Automation
+{
+    public class ApiPreValidation : EntityPreValidationBase, IValidation
+    {
+        public ApiPreValidation(string folderPath)
+            : base(folderPath)
+        {
+        }
+
+        public Task<bool> ValidateAsync(GatewayDetails gatewayDetails)
```

Comment:
If `ApiPreValidation` validates Api, why it requires whole `GatewayDetails` object?

Reply:
It is common interface signature, which will be used for other pre validation

Reply:
Ok. Let's see how it's going to work.